compiler/rustc_errors/src/json.rs

```diff
@@ -85,6 +89,7 @@ impl JsonEmitter {
             macro_backtrace: false,
             track_diagnostics: false,
             terminal_url: TerminalUrl::No,
+            start_timestamp: Instant::now(),
```

Comment:
Is this the starting timestamp against which all emitted durations are measured? The assumption is there is a single `JsonEmitter`? There's actually an "early" one as well as the main one (search for `JsonEmitter::new()` calls) but maybe that doesn't matter. Also, we're assuming this `JsonEmitter` is created close enough to start-up that it's as good as "time zero"?

Reply:
Yes, the assumption is that there is a single emitter, I think the earlier ones are only used for early diagnostics and they are not used for the actual compilation (in other words, we probably don't have any interesting sections before the "main" JSON emitter is created). The emitter doesn't need to be created close to start up, I wouldn't promise that the time emitted is "since the start of the rustc process", that's on the user to compute.

In fact, due to https://github.com/rust-lang/rust/pull/142123#discussion_r2135220410, do you think we need to emit any timestamps at all? Or maybe we could just emit the UNIX timestamp :man_shrugging:

Reply:
I see. That probably won't be super relevant for the timings. But in any case, using UNIX timestamp seems like a simpler option.